### PR TITLE
Limit io to callbacks, sync profiles and timeseries (v2)

### LINKED ISSUE #821: Bad values in the last element of timeseries output
I'm getting garbage as a last element of the timeseries output data. 

For Dycoms_RF02 case that I'm running, for example

`cb_tc = np.array(ds_tc['timeseries']['cloud_base_mean'])`
`print(cb_tc[-1], cb_tc[-2])`

gives
`9.969209968386869e+36 375.0`

It's the same for a couple of other timeseries outputs that I checked. 

So far I have checked that the simulation itself is fine, i.e. it's not actually about to crash. It really seems to just be the output problem. I will be looking into it, but any help from more output savvy people is appreciated.

### PR DESCRIPTION
**Pull request by ilopezgp**
- Synchronize first io call within main for all profiles and time series.
- Fixes #821 and #897.
- Tests for filled values in integration tests, which were not covered previously (hence the errors).
- Initializes integrator during initialization, and passes it as an argument to `run`.

### CODE REVIEW

**Comment by ilopezgp** [comment]
Would this work @charleskawczynski ? Dealing with it through ODE is nasty

**Comment by ilopezgp** [comment]
All cases seem to work, except for Bomex on the sphere.

File: driver/main.jl [hunk 1]
```diff
@@ -252,13 +252,13 @@ function initialize(sim::Simulation1d)
...
+        (prob, alg, kwargs) = solve_args(sim)
+        integrator = ODE.init(prob, alg; kwargs...)
+
         open_files(stats)
         try
-            write_simulation_time(stats, t)
-            io(io_nt.aux, stats, state)
-            io(io_nt.diagnostics, stats, diagnostics_col)
-            io(surf, case.surf_params, grid, state, stats, t)
+            affect_io!(integrator)
```
Comment (charleskawczynski) [inline driver/main.jl:261]: Affect up loops over columns, so we don’t want to do it like this.

**Comment by charleskawczynski** [comment]
What was the issue with doing this through callbacks?

**Comment by ilopezgp** [comment]
> What was the issue with doing this through callbacks?

I did a deep dive into callbacks with ODE and bottomline, the package does not handle well callbacks at initialization. I tried save_start, save_at, different callbacks... None worked as I expected

**Comment by ilopezgp** [comment]
@charleskawczynski I fixed the column loop and added tests for fillvalues, so we know we are not making the same mistake as before.

**Review by charleskawczynski — COMMENTED** [review]
I think this is close.

I think we can avoid initializing the integrator twice by making an `affect_io` wrapper and calling it with the raw components in ‘initialize(::Simulation1D)`.

**Comment by ilopezgp** [comment]
> I think this is close.
> 
> I think we can avoid initializing the integrator twice (which is expensive and could cause confusion with print debugging) by making an `affect_io` wrapper and calling it with the raw components in ‘initialize(::Simulation1D)`.
> 
> If what I said doesn’t make sense I can try to add what I was thinking
> 
> also, thanks for adding tests!!

I can just pass the integrator downstream

File: driver/main.jl [hunk 2]
```diff
@@ -252,26 +252,26 @@ function initialize(sim::Simulation1d)
...
+        else
+            @warn "IO during initialization failed." exception = (e, catch_backtrace())
         end
+        return :simulation_crashed
+    finally
+        close_files(sim)
     end
 
-    return
+    return integrator
```
Comment (charleskawczynski) [inline driver/main.jl:274]:
```suggestion
    return (integrator, :success)
```
And catch / use downstream

**Comment by charleskawczynski** [comment]
> I can just pass the integrator downstream

Yeah, I think that should work fine, too

**Comment by ilopezgp** [comment]
I think this is ready @charleskawczynski

File: driver/main.jl [hunk 3]
```diff
@@ -397,11 +397,12 @@ function main1d(namelist; time_run = true)
         end
     end
     sim = Simulation1d(namelist)
-    initialize(sim)
+    (integrator, return_init) = initialize(sim)
+    return_init == :success && @info "The initialization has completed."
```
Comment (charleskawczynski) [inline driver/main.jl:401]: What should we do if `return_init` is `:simulation_crashed`?

Comment (ilopezgp) [reply driver/main.jl:401]: I guess same as before, nothing? It will break when it starts running

**Review by charleskawczynski — APPROVED** [review]
Not a blocker, but we may want to more carefully handle the case when `return_init` is `:simulation_crashed` (this is probably more important for CEDMF).

File: driver/main.jl (hunk 3 above)
Comment (charleskawczynski) [reply driver/main.jl:401]: It's not totally clear to me that that will be the case, but you're probably right in 99% of the cases. Again, not a blocker.

**Comment by ilopezgp** [comment]
> Not a blocker, but we may want to more carefully handle the case when `return_init` is `:simulation_crashed` (this is probably more important for CEDMF).

I will implement this solution in theta-var and report back if this results in unexpected behavior during calibration.

**Pull request merged**